src/CentralApi/Services/RobotService.cs

```diff
@@ -0,0 +1,227 @@
+﻿using Middleware.CentralApi.Services.Abstract;
+using Middleware.DataAccess.Repositories.Abstract;
+using Middleware.Models.Domain;
+using Middleware.CentralApi.Contracts.Responses;
+using Middleware.Common.Enums;
+using KeyValuePair = Middleware.Models.Domain.KeyValuePair;
+
+namespace Middleware.CentralApi.Services;
+
+public class RobotService : IRobotService
+{
+    private readonly ILogger _logger;
+    private readonly IRobotRepository _robotRepository;
+    private readonly ILocationRepository _locationRepository;
+
+    public RobotService(ILogger<RobotService> logger,
+        IRobotRepository robotRepository,
+        ILocationRepository locationRepository)
+    {
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _robotRepository = robotRepository ?? throw new ArgumentNullException(nameof(robotRepository));
+        _locationRepository = locationRepository ?? throw new ArgumentNullException(nameof(locationRepository));
+    }
+
+    public async Task <List<string>>CreateRelation(RelationToLocationRequest data)
+    {
+        List<LocationNames> relGraph2s = data.Locations;
+        // check if name of robot can be retreived
+        var robotName = await GetRobotNameByIdAsync(data.robotId);
+        if (robotName == null)
+            throw new ArgumentNullException(nameof(robotName));
+        
+
+        List<string> errorsList = new List<string>();
+        List<RelationModel> availableRrelations = new();
+        var relationName = "CAN_REACH";
+
+        availableRrelations = await _robotRepository.GetRelation(data.robotId, relationName);
+
+        foreach (LocationNames relGraph2 in relGraph2s)
+        {
+            if (relGraph2 != null)
+            {
+                var organisationName = relGraph2.OrganisationName;
+                var locationName = relGraph2.LocationName;
+
+                if (organisationName != null && organisationName != string.Empty && locationName != null && locationName != string.Empty)
```

Comment:
You can simplify this `if` statement by using 
`!string.IsNullOrEmpty(organizationName) && !string.IsNullOrEmpty(locationName)`